pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -0,0 +1,256 @@
+package triggers
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/mercury"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// TODO: Register the capability in the NewDelegateFunction in chainlink/core/services/workflows.delegate.go
+
+var mercuryInfo = capabilities.MustNewCapabilityInfo(
+	"mercury-trigger",
+	capabilities.CapabilityTypeTrigger,
+	"An example mercury trigger.",
+	"v1.0.0",
+)
+
+// This Trigger Service allows for the registration and deregistration of triggers. You can also send reports to the service.
+type MercuryTriggerService struct {
+	capabilities.CapabilityInfo
+	chans                 map[string]chan<- capabilities.CapabilityResponse
+	feedIDsForTriggerID   map[string][]int64 // TODO: switch this to uint64 when value.go supports it
+	triggerIDsForFeedID   map[int64][]string
+	mu                    sync.Mutex
+	triggerIDToWorkflowID map[string]string
+}
+
+var _ capabilities.TriggerCapability = (*MercuryTriggerService)(nil)
+
+func NewMercuryTriggerService() *MercuryTriggerService {
+	return &MercuryTriggerService{
+		CapabilityInfo:        mercuryInfo,
+		chans:                 map[string]chan<- capabilities.CapabilityResponse{},
+		feedIDsForTriggerID:   make(map[string][]int64),
+		triggerIDsForFeedID:   make(map[int64][]string),
+		triggerIDToWorkflowID: make(map[string]string),
+	}
+}
+
+// maybe use mercury-pipline reports.go report struct instead of MercuryReport struct, and then we can use the pipeline report generators
```

Comment:
This comment is confusing, maybe remove it and put it in some follow-up JIRA ticket with more details?